google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/ConnectionStatementExecutorImpl.java

```diff
@@ -0,0 +1,227 @@
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SET_READ_ONLY_STALENESS;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SET_RETRY_ABORTS_INTERNALLY;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SET_STATEMENT_TIMEOUT;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SET_TRANSACTION_MODE;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_AUTOCOMMIT;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_AUTOCOMMIT_DML_MODE;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_COMMIT_TIMESTAMP;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_READONLY;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_READ_ONLY_STALENESS;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_READ_TIMESTAMP;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_RETRY_ABORTS_INTERNALLY;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.SHOW_STATEMENT_TIMEOUT;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.START_BATCH_DDL;
+import static com.google.cloud.spanner.jdbc.StatementResult.ClientSideStatementType.START_BATCH_DML;
+import static com.google.cloud.spanner.jdbc.StatementResultImpl.noResult;
+import static com.google.cloud.spanner.jdbc.StatementResultImpl.resultSet;
+import java.util.concurrent.TimeUnit;
+import com.google.cloud.spanner.TimestampBound;
+import com.google.cloud.spanner.jdbc.ReadOnlyStalenessUtil.DurationValueGetter;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.Duration;
+
+/**
+ * The methods in this class are called by the different {@link ClientSideStatement}s. These method
+ * calls are then forwarded into a {@link Connection}.
+ */
+class ConnectionStatementExecutorImpl implements ConnectionStatementExecutor {
+  static final class StatementTimeoutGetter implements DurationValueGetter {
+    private final Connection connection;
+
+    public StatementTimeoutGetter(Connection connection) {
+      this.connection = connection;
+    }
+
+    @Override
+    public long getDuration(TimeUnit unit) {
+      return connection.getStatementTimeout(unit);
+    }
+
+    @Override
+    public boolean hasDuration() {
+      return connection.hasStatementTimeout();
+    }
+  }
+
+  /** The connection to execute the statements on. */
+  private final ConnectionImpl connection;
+
+  ConnectionStatementExecutorImpl(ConnectionImpl connection) {
+    this.connection = connection;
+  }
+
+  ConnectionImpl getConnection() {
+    return connection;
+  }
+
+  @Override
+  public StatementResult statementSetAutocommit(Boolean autocommit) {
```

Comment:
should autocommit be autoCommit? WDYT?

Reply:
(not sure if this is an already defined naming convention elsewhere)

Reply:
Both are used. I have a slight preference for `autocommit`, but I don't feel very strongly about this.

JDBC uses `autoCommit`. [SQL Server](https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/auto-commit-mode?view=sql-server-2017) uses `auto-commit`. [PostgreSQL](https://www.postgresql.org/docs/11/ecpg-sql-set-autocommit.html) and [MySQL](https://dev.mysql.com/doc/refman/8.0/en/innodb-autocommit-commit-rollback.html) use `autocommit` in their documentation.

Reply:
One reason to keep this as `autocommit` is that we have defined the variable `AUTOCOMMIT_DML_MODE`, which indicates that the driver assumes autocommit to be one word.

Reply:
SGTM